packages/web3-providers-http/src/index.js

```diff
@@ -111,6 +111,10 @@ HttpProvider.prototype.send = function (payload, callback) {
         }
     };
 
+    request.onerror = function() {
+        callback(errors.RequestFailed());
```

Comment:
Would probably be great to get our users some more information about the failed requests. :-)

Reply:
Agreed :) 

I wrote a little bit about this towards the end of the PR description, would you be open to having `XMLHttpRequest` being replaced with another library like `fetch` as part of this PR?